ports/mesa/portfile.cmake

```
@@ -0,0 +1,171 @@
+list(APPEND MESA_OPTIONS -Dzstd=enabled)
+list(APPEND MESA_OPTIONS -Dshared-llvm=auto)
+list(APPEND MESA_OPTIONS -Dlibunwind=disabled)
+list(APPEND MESA_OPTIONS -Dlmsensors=disabled)
+list(APPEND MESA_OPTIONS -Dvalgrind=disabled)
+list(APPEND MESA_OPTIONS -Dglvnd=false)
+list(APPEND MESA_OPTIONS -Dglx=disabled)
+list(APPEND MESA_OPTIONS -Dgbm=disabled)
+#list(APPEND MESA_OPTIONS -Dosmesa=['gallium','classic']) # classic has compiler errors. 
+list(APPEND MESA_OPTIONS -Dosmesa=['gallium'])
+
+
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    list(APPEND MESA_OPTIONS -Dshared-swr=false)
+    list(APPEND MESA_OPTIONS "-Dswr-arches=['avx']")
+else()
+    list(APPEND MESA_OPTIONS -Dshared-swr=true)
+    list(APPEND MESA_OPTIONS "-Dswr-arches=['avx','avx2','knl','skx']")
+endif()
+
+string(APPEND GALLIUM_DRIVERS 'swrast')
+if("llvm" IN_LIST FEATURES)
+    list(APPEND MESA_OPTIONS -Dllvm=enabled)
+    string(APPEND GALLIUM_DRIVERS ",'swr'") # SWR always requires llvm
+else()
+    list(APPEND MESA_OPTIONS -Dllvm=disabled)
+endif()
+
+list(APPEND MESA_OPTIONS -Dgallium-drivers=[${GALLIUM_DRIVERS}])
+
+if("gles1" IN_LIST FEATURES)
+    list(APPEND MESA_OPTIONS -Dgles1=enabled)
+else()
+    list(APPEND MESA_OPTIONS -Dgles1=disabled)
+endif()
+if("gles2" IN_LIST FEATURES)
+    list(APPEND MESA_OPTIONS -Dgles2=enabled)
+else()
+    list(APPEND MESA_OPTIONS -Dgles2=disabled)
+endif()
+if("opengl" IN_LIST FEATURES)
+    list(APPEND MESA_OPTIONS -Dopengl=true)
+else()
+    list(APPEND MESA_OPTIONS -Dopengl=false)
+endif()
+if("egl" IN_LIST FEATURES) # EGL feature only works on Linux
+    list(APPEND MESA_OPTIONS -Degl=enabled)
+else()
+    list(APPEND MESA_OPTIONS -Degl=disabled)
+endif()
+
+list(APPEND MESA_OPTIONS -Dshared-glapi=enabled)  #shared GLAPI required when building two or more of the following APIs - opengl, gles1 gles2
+
+
+if(VCPKG_TARGET_IS_WINDOWS)
+    list(APPEND MESA_OPTIONS -Dplatforms=['windows'])
+endif()
+
+vcpkg_configure_meson(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS 
+        #-D gles-lib-suffix=_mesa
+        #-D egl-lib-suffix=_mesa
+        -Dbuild-tests=false
+        ${MESA_OPTIONS}
+    )
+vcpkg_install_meson()
+vcpkg_fixup_pkgconfig(SYSTEM_LIBRARIES pthread)
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
+
+#installed by egl-registry
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/include/KHR)
+file(REMOVE ${CURRENT_PACKAGES_DIR}/include/EGL/egl.h)
+file(REMOVE ${CURRENT_PACKAGES_DIR}/include/EGL/eglext.h)
+file(REMOVE ${CURRENT_PACKAGES_DIR}/include/EGL/eglplatform.h)
+
+#installed by opengl-registry
+set(_double_files include/GL/glcorearb.h include/GL/glext.h include/GL/glxext.h 
+    include/GLES/egl.h include/GLES/gl.h include/GLES/glext.h include/GLES/glplatform.h 
+    include/GLES2/gl2.h include/GLES2/gl2ext.h include/GLES2/gl2platform.h
+    include/GLES3/gl3.h  include/GLES3/gl31.h include/GLES3/gl32.h include/GLES3/gl3platform.h)

```

```suggestion
set(_double_files include/GL/glcorearb.h include/GL/glext.h include/GL/glxext.h
    include/GLES3/gl3.h  include/GLES3/gl31.h include/GLES3/gl32.h include/GLES3/gl3platform.h)
```
Below is already removing the `GLES` and `GLES2` dirs.